rudder-core/src/main/scala/com/normation/rudder/ncf/Technique.scala

```diff
@@ -0,0 +1,78 @@
+/*
+*************************************************************************************
+* Copyright 2017 Normation SAS
+*************************************************************************************
+*
+* This file is part of Rudder.
+*
+* Rudder is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* In accordance with the terms of section 7 (7. Additional Terms.) of
+* the GNU General Public License version 3, the copyright holders add
+* the following Additional permissions:
+* Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+* Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+* Public License version 3, when you create a Related Module, this
+* Related Module is not considered as a part of the work and may be
+* distributed under the license agreement of your choice.
+* A "Related Module" means a set of sources files including their
+* documentation that, without modification of the Source Code, enables
+* supplementary functions or services in addition to those offered by
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+
+package com.normation.rudder.ncf
+
+import com.normation.inventory.domain.Version
+import com.normation.inventory.domain.AgentType
+
+case class BundleName (value : String) {
+  val toDsc = value.split("_").map(_.capitalize).mkString("-")
```

Comment:
I think that should be externalized in an obiect ("Encoder?) with a more generic name about what is the goal of the method (toHumanName ?)

Reply:
What to you think of the way i choose ?

Reply:
Regrouping the case clases into a trait, then adding an implciit class to have the encoder method when needed

I think i can regroup BundleName and ParameterId into a single case class ... but having two separate case class allow to have different semantic on the classe (which here is interesting, since i don't want to use a BundleName as a Key for my parameters ... ) but they have the same transformations and conditions